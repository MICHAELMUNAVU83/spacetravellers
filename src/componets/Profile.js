import React from 'react';
import { useSelector } from 'react-redux';

function Profile() {
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  const newRockets = reservedRockets.map((reservedRocket) => (
    <tbody key={reservedRocket.id}>
      <tr>
        <td>
          {' '}
          {reservedRocket.rocket_name}
        </td>
      </tr>
    </tbody>

  ));
  const dragons = useSelector((state) => state.dragons);
  const reservedDragons = dragons.filter((dragon) => dragon.reserved);
  const newDragons = reservedDragons.map((reservedDragon) => (
    <tbody key={reservedDragon.id}>
      <tr>
        <td>
          {' '}
          {reservedDragon.dragon_name}
        </td>
      </tr>
    </tbody>
  ));
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.join);
  const newMissions = reservedMissions.map((reservedMission) => (
    <tbody key={reservedMission.id}>
      <tr>
        <td>
          {' '}
          {reservedMission.mission_name}
        </td>
      </tr>
    </tbody>
  ));

  return (
    <div className="profiles">
      <div>
        <h2>MY ROCKETS</h2>
        <table className="table table-bordered">{newRockets}</table>
      </div>
      <div>
        <h2>MY DRAGONS</h2>
        <table className="table table-bordered">{newDragons}</table>
      </div>
      <div>
        <h2>MY MISSIONS</h2>
        <table className="table table-bordered">{newMissions}</table>
      </div>
    </div>
  );
}

export default Profile;
