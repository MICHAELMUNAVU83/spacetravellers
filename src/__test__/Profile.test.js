import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../componets/Profile';
import store from '../redux/configureStore';

it('Profile renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Profile />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
