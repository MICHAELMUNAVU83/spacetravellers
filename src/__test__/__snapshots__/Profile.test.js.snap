// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="profiles"
>
  <div>
    <h2>
      MY ROCKETS
    </h2>
    <table
      className="table table-bordered"
    />
  </div>
  <div>
    <h2>
      MY DRAGONS
    </h2>
    <table
      className="table table-bordered"
    />
  </div>
  <div>
    <h2>
      MY MISSIONS
    </h2>
    <table
      className="table table-bordered"
    />
  </div>
</div>
`;
